### DISCUSSION TITLE
Updating row data for two tables with the same submit, with initial data pulled from a data provider...

### DISCUSSION THREAD

**Discussion by cfepdan** [Budibase/budibase#5051] [Help]
This is similar to #4461 which discusses using a single submit button to create rows from two fields in two separate tables.

This is good, but I want to _update_ a row and its related row in another table with a single submit. I'd like to see the original data in the fields before updating. If there were just one table, I would nest the field within a form within a repeat within a data provider... But with two tables, and without the data being in the related field object, I don't seem to be able to do this. Any hints?

**Comment by cfepdan** [comment]
I *think* I can muster this by using a SQL database and having the data provider use a join query. But if there's a way to do this using the built-in db, lmk please.

**Reply by mjashanks** [reply]
Hey @cfepdan apologies for the delay. 

You should be able to do this by adding a second datasource, without a repeater

```
> Data Provider 1 - table 1
  > Data Provider 2 - table 2
  > Repeater (table 1)
     >  ... repeated components...
```

The Repeater will do it's think with table 1. However, you can still access rows on table 2 by doing a binding like

`{{ Data Provider 2.Rows.0.My Column Name }}`

The same works in Javascript.

**Reply by cfepdan** [reply]
I like this. I'll give it a shot. For the moment, it made sense to put all the stuff I had in the other table into a JSON object instead. There wasn't a reason to pull it from another table other than to keep the column count down on my parent table.
